src/main/java/org/truffleruby/language/objects/classvariables/ClassVariableStorage.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 2.0, or
+ * GNU General Public License version 2, or
+ * GNU Lesser General Public License version 2.1.
+ */
+package org.truffleruby.language.objects.classvariables;
+
+import com.oracle.truffle.api.object.DynamicObject;
+import com.oracle.truffle.api.object.Shape;
+
+public class ClassVariableStorage extends DynamicObject {
+
+    private static final Shape INITIAL_SHAPE = Shape.newBuilder().layout(ClassVariableStorage.class).build();
```

Comment:
Shapes should be stored in RubyLanguage, otherwise we might get type pollution between independent programs run in different engines.